crates/ruma-client-api/src/sync/sync_events/v4.rs

```diff
@@ -333,3 +366,127 @@ impl SlidingSyncRoom {
         Default::default()
     }
 }
+
+/// Configure Sliding-Sync extensions.
+#[derive(Clone, Debug, Default, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct ExtensionsRequest {
+    /// Request to devices messages with the given config.
+    pub to_device: Option<ToDeviceRequest>,
+    /// Configure the end-to-end-encryption extension.
+    pub e2ee: Option<E2EERequest>,
+    /// Configure the account data extension.
+    pub account_data: Option<AccountDataRequest>,
+    /// Extensions may add further fields to the list.
+    #[serde(flatten, default, skip_serializing_if = "BTreeMap::is_empty")]
+    pub other: BTreeMap<String, serde_json::Value>,
```

Comment:
```suggestion
    /// Request to devices messages with the given config.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub to_device: Option<ToDeviceRequest>,

    /// Configure the end-to-end-encryption extension.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub e2ee: Option<E2EERequest>,

    /// Configure the account data extension.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub account_data: Option<AccountDataRequest>,

    /// Extensions may add further fields to the list.
    #[serde(flatten)]
    other: BTreeMap<String, serde_json::Value>,
```

Regarding the attribute change on `other`, these attributes are no-ops when combined with flatten AFAIK. The visibility is sth. I think we discussed before, but feel free to ping me on Matrix about it.